### PR TITLE
Add post grad graph logging

### PR DESCRIPTION
**Pull request by oulgen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111808
* #111770

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_logging/_registrations.py [hunk 1]
```diff
@@ -45,6 +45,11 @@
     "aot_joint_graph",
     "Print FX joint graph from AOTAutograd, prior to partitioning. Useful for debugging partitioning",
 )
+register_artifact(
+    "post_grad_graphs",
+    "Prints the FX graph generated by post grad passes. Useful to understand what's being given to Inductor after post grad passes",
+    visible=True,
```
Comment (Chillee) [inline torch/_logging/_registrations.py:51]: Do we want to make it visible by default (perhaps that's a bad name)? That just shows it by default when you do `TORCH_LOGS="help"`

**Review by Chillee — APPROVED** [review]

File: torch/_logging/_registrations.py (hunk 1 above)
Comment (oulgen) [reply torch/_logging/_registrations.py:51]: output_code and aot_graphs are visible, and this is no different. I can change it if this isn't as important or as useful.

Comment (Chillee) [reply torch/_logging/_registrations.py:51]: Yeah it's filtered by "how useful do we expect this to be to the average PyTorch user".

This feels like a bit more of a dev detail, so perhaps better to not show it by default. To be clear, this has no impact on the actual usage/what's shown, it's only about what's shown in the help message.

**Pull request closed without merging**